source/opt/dead_branch_elim_pass.cpp

```diff
@@ -220,6 +220,7 @@ bool DeadBranchElimPass::EliminateDeadBranches(ir::Function* func) {
     const uint32_t mergeLabId =
         mergeInst->GetSingleWordInOperand(kSelectionMergeMergeBlockIdInIdx);
     AddBranch(liveLabId, *bi);
+    backedges_.erase(br);
```

Comment:
I think this may never be a backedge, so I am worried it could be confusing. I think it is safe, so whatever you think best here.

Reply:
I think this case occurs for a single-block loop.  Granted, Glslang doesn't generate loops that way, so for now those are typically hand-constructed cases.

But this merits adding a test case, and perhaps a comment as well.  I'll do that in another PR.